src/Engine/TranskribusClient.php

```diff
@@ -137,8 +146,52 @@ private function request(
                 return $content;
             }
             $this->throwException(0);
-        } else {
-            $this->throwException($statusCode);
         }
+
+        if (401 === $statusCode) {
+            $this->refreshAccessToken();
+
+            $body['headers']['Authorization'] = 'Bearer '.$this->accessToken;
+            $response = $this->httpClient->request($method, $url, $body);
+            $statusCode = $response->getStatusCode();
+
+            if (200 === $statusCode) {
+                $content = json_decode($response->getContent());
+
+                if (!empty($content)) {
+                    return $content;
+                }
+                $this->throwException(0);
+            }
+        }
+        $this->throwException($statusCode);
+    }
+
+    private function refreshAccessToken(): void
+    {
+        $response = $this->httpClient->request(
```

Comment:
This request is identical to the one done in `TranskribusAuthCommand` isn't it? I wonder if it'd be neater to inject a `TranskribusClient` into the command, and so only have this code in one place?

Reply:
New static methods have been added.

Reply:
The method shouldn't be static. A TranskribusClient object should be injected in the constructor of the command, and then the method used non-statically. Does that make sense?

Reply:
Static methods to avoid injecting empty accessToken/refereshToken when running the command for the first time. Updates have been made. The methods are no longer static. The client no longer requires the accesstoken and refreshtoken when it's created.